Adds "Recent Incidents Only" functionality

binarygeotech opened the pull request:
Close #144
* Adds settings to show only recent incidents by adding new fields to the settings page.
  * Show Recent Incidents Only (toggle button)
  * Number of Days to Show Recent Incidents (number input)
* Disables date filter on the incidents page if "Show Recent Incidents Only" is true
* Hides the pagination links on the incidents page if "Show Recent Incidents Only" is true
* Applies filter to `incidents` query based on the specified number of days on the settings page
* Applies filter to `incidents` query for RSS feeds based on the specified number of days on the settings page

binarygeotech commented:
@jbrooksuk kindly take a look at this PR if it does what you proposed on the referenced issue.

jbrooksuk commented on src/View/Components/IncidentTimeline.php:
```diff
@@ -33,7 +35,10 @@ public function render(): View
-            'canPageForward' => $startDate->clone()->isBefore(now()),
+            'canPageForward' => $this->appSettings->recent_incidents_only ? false : $startDate->clone()->isBefore(now()),
+            'canPageBackward' => $this->appSettings->recent_incidents_only ? false : true,
+            'recent_incidents_only' => $this->appSettings->recent_incidents_only,
```
Can the variables be camelCase please.

jbrooksuk commented on src/View/Components/IncidentTimeline.php:
```diff
@@ -43,21 +48,32 @@ public function render(): View
      */
     private function incidents(Carbon $startDate, Carbon $endDate, bool $onlyDisruptedDays = false): Collection
     {
+        $appSettings = $this->appSettings;
```
Since we need to pass this variable to lower scopes, I'd probably avoid the temporary variable and use the property within the closure.

jbrooksuk commented:
By the way, @binarygeotech we'll also need to update the translation file.

binarygeotech commented:
@jbrooksuk your suggested updates have been applied;
* Updated variable name to camelCase
* Updated the translation file
* Removed redundant local variable `$appSettings`
* Removed unused variable `recentIncidentsDays` from being passed to the UI

jbrooksuk commented:
Thanks! Code looks good to me, I'm not at my laptop to test yet, but I'll be back soon.

jbrooksuk commented on src/View/Components/IncidentTimeline.php:
```diff
@@ -49,15 +53,24 @@ private function incidents(Carbon $startDate, Carbon $endDate, bool $onlyDisrupt
-                ])->orWhere(function (Builder $query) use ($endDate, $startDate) {
-                    $query->whereNull('occurred_at')->whereBetween('created_at', [
+            ->when($this->appSettings->recent_incidents_only, function ($query) {
+                $query->whereDate(
+                    'occurred_at',
+                    '>',
+                    Carbon::now()->subDays($this->appSettings->recent_incidents_days)->format('Y-m-d')
```
I think this also needs to account for the `created_at` timestamp, when `occurred_at` is null.

binarygeotech replied:
Ok, I will make necessary adjustments.

binarygeotech commented:
@jbrooksuk the required update have been made, kindly review. Happy new year to you from Nigeria.

binarygeotech commented:
@jbrooksuk I've update the PR with the recent localisation updates. I believe the PR is good to go now.

jbrooksuk commented:
This is awesome, thanks for your help on this feature, @binarygeotech!
